lib/web_ui/dev/safari_macos.dart

```diff
@@ -23,6 +24,58 @@ class SafariMacOsEnvironment extends WebDriverBrowserEnvironment {
   @override
   Uri get driverUri => Uri(scheme: 'http', host: 'localhost', port: portNumber);
 
+  late Process _driverProcess;
+  int _retryCount = 0;
+  static const int _waitBetweenRetryInSeconds = 1;
+  static const int _maxRetryCount = 10;
+
   @override
   Future<Process> spawnDriverProcess() => Process.start('safaridriver', <String>['-p', portNumber.toString()]);
+
+  @override
+  Future<void> prepare() async {
+    await _startDriverProcess();
+  }
+
+  /// Pick an unused port and start `safaridriver` using that port.
+  ///
+  /// On macOS 13, starting `safaridriver` can be flaky so if it returns an
+  /// "Operation not permitted" error, kill the `safaridriver` process and try
+  /// again with a different port. Wait [_waitBetweenRetryInSeconds] seconds
+  /// between retries. Try up to [_maxRetryCount] times.
+  Future<void> _startDriverProcess() async {
```

Comment:
Is there a reason this isn't just the implementation of `spawnDriverProcess` rather than overriding `prepare`?

Reply:
I see. Thanks for the clarification!